File: app/src/main/java/com/mmjang/ankihelper/data/ai/AIManager.java
package com.mmjang.ankihelper.data.ai;

import android.content.Context;
import android.widget.Toast;

import com.mmjang.ankihelper.MyApplication;
import com.mmjang.ankihelper.data.ai.cache.AIDictionaryCache;
import com.mmjang.ankihelper.data.ai.service.AIDictionaryService;
import com.mmjang.ankihelper.data.ai.service.AITranslatorService;

import java.io.IOException;
import java.util.List;

public class AIManager {
    private static AIManager instance;
    private AIDictionaryService dictionaryService;
    private AITranslatorService translatorService;
    
    private AIManager() {
        this.dictionaryService = new AIDictionaryService();
        this.translatorService = new AITranslatorService();
    }
    
    public static synchronized AIManager getInstance() {
        if (instance == null) {
            instance = new AIManager();
        }
        return instance;
    }
    
    public List<AIDictionaryCache> getWordDefinition(String word, long dictionaryConfigId) throws IOException, AIException {
        // Check network connectivity
        if (!NetworkUtil.isNetworkAvailable(MyApplication.getContext())) {
            throw new IOException("No network connection. Please check your internet connection and try again.");
        }
        
        AIDictionaryConfig dictionaryConfig = AIConfigRepository.getAIDictionaryConfigById(dictionaryConfigId);
        if (dictionaryConfig == null) {
            throw new IOException("AI Dictionary configuration not found");
        }
        
        LLMConfig llmConfig = AIConfigRepository.getLLMConfigById(dictionaryConfig.getLlmId());
        if (llmConfig == null) {
            throw new IOException("LLM configuration not found");
        }
        
        return dictionaryService.getWordDefinition(word, dictionaryConfig, llmConfig);
    }
    
    public String translateText(String text, long translatorConfigId) throws IOException, AIException {
        // Check network connectivity
        if (!NetworkUtil.isNetworkAvailable(MyApplication.getContext())) {
            throw new IOException("No network connection. Please check your internet connection and try again.");
        }
        
        AITranslatorConfig translatorConfig = AIConfigRepository.getAITranslatorConfigById(translatorConfigId);
        if (translatorConfig == null) {
            throw new IOException("AI Translator configuration not found");
        }
        
        LLMConfig llmConfig = AIConfigRepository.getLLMConfigById(translatorConfig.getLlmId());
        if (llmConfig == null) {
            throw new IOException("LLM configuration not found");
        }
        
        return translatorService.translateText(text, translatorConfig.getSourceLanguage(), 
                translatorConfig.getTargetLanguage(), translatorConfig, llmConfig);
    }
    
    public String translateTextWithDefaultTranslator(String text) throws IOException, AIException {
        // Check network connectivity
        if (!NetworkUtil.isNetworkAvailable(MyApplication.getContext())) {
            throw new IOException("No network connection. Please check your internet connection and try again.");
        }
        
        AITranslatorConfig translatorConfig = AIConfigRepository.getDefaultAITranslatorConfig();
        if (translatorConfig == null) {
            throw new IOException("No default AI Translator configuration found");
        }
        
        LLMConfig llmConfig = AIConfigRepository.getLLMConfigById(translatorConfig.getLlmId());
        if (llmConfig == null) {
            throw new IOException("LLM configuration not found");
        }
        
        return translatorService.translateText(text, translatorConfig.getSourceLanguage(), 
                translatorConfig.getTargetLanguage(), translatorConfig, llmConfig);
    }
}

File: app/src/main/java/com/lmyby/ankiquicker/data/ai/AIManager.kt
package com.lmyby.ankiquicker.data.ai

import com.lmyby.ankiquicker.MyApplication
import com.lmyby.ankiquicker.data.ai.cache.AIDictionaryCache
import com.lmyby.ankiquicker.data.ai.service.AIDictionaryService
import com.lmyby.ankiquicker.data.ai.service.AITranslatorService
import java.io.IOException

/**
 * Converted to Kotlin as part of Phase 2 data model migration
 */
class AIManager private constructor() {
    private val dictionaryService: AIDictionaryService = AIDictionaryService()
    private val translatorService: AITranslatorService = AITranslatorService()

    @Throws(IOException::class, AIException::class)
    fun getWordDefinition(word: String, dictionaryConfigId: Long): List<AIDictionaryCache> {
        // Check network connectivity
        if (!NetworkUtil.isNetworkAvailable(MyApplication.getContext())) {
            throw IOException("No network connection. Please check your internet connection and try again.")
        }

        val dictionaryConfig = AIConfigRepository.getAIDictionaryConfigById(dictionaryConfigId)
            ?: throw IOException("AI Dictionary configuration not found")

        val llmConfig = AIConfigRepository.getLLMConfigById(dictionaryConfig.llmId)
            ?: throw IOException("LLM configuration not found")

        return dictionaryService.getWordDefinition(word, dictionaryConfig, llmConfig)
    }

    @Throws(IOException::class, AIException::class)
    fun translateText(text: String, translatorConfigId: Long): String {
        // Check network connectivity
        if (!NetworkUtil.isNetworkAvailable(MyApplication.getContext())) {
            throw IOException("No network connection. Please check your internet connection and try again.")
        }

        val translatorConfig = AIConfigRepository.getAITranslatorConfigById(translatorConfigId)
            ?: throw IOException("AI Translator configuration not found")

        val llmConfig = AIConfigRepository.getLLMConfigById(translatorConfig.llmId)
            ?: throw IOException("LLM configuration not found")

        return translatorService.translateText(
            text,
            translatorConfig.sourceLanguage ?: "auto",
            translatorConfig.targetLanguage ?: "en",
            translatorConfig,
            llmConfig
        )
    }

    @Throws(IOException::class, AIException::class)
    fun translateTextWithDefaultTranslator(text: String): String {
        // Check network connectivity
        if (!NetworkUtil.isNetworkAvailable(MyApplication.getContext())) {
            throw IOException("No network connection. Please check your internet connection and try again.")
        }

        val translatorConfig = AIConfigRepository.getDefaultAITranslatorConfig()
            ?: throw IOException("No default AI Translator configuration found")

        val llmConfig = AIConfigRepository.getLLMConfigById(translatorConfig.llmId)
            ?: throw IOException("LLM configuration not found")

        return translatorService.translateText(
            text,
            translatorConfig.sourceLanguage ?: "auto",
            translatorConfig.targetLanguage ?: "en",
            translatorConfig,
            llmConfig
        )
    }

    companion object {
        @Volatile
        private var instance: AIManager? = null

        @JvmStatic
        fun getInstance(): AIManager {
            return instance ?: synchronized(this) {
                instance ?: AIManager().also { instance = it }
            }
        }
    }
}
